Allows to bind to any port, as specified by PORT env var, not just 53.
Should be applied on top of fefe.de IPv6 test27 patches.

Source: github.com/mk-fg/archlinux-pkgbuilds

---
diff --git a/Makefile b/Makefile
index 9ebf4c8..a077d23 100644
--- a/Makefile
+++ b/Makefile
@@ -837,7 +837,7 @@ choose compile trysysel.c select.h1 select.h2
 	./choose c trysysel select.h1 select.h2 > select.h
 
 server.o: \
-compile server.c byte.h case.h env.h buffer.h strerr.h ip4.h uint16.h \
+compile server.c byte.h case.h env.h scan.h buffer.h strerr.h ip4.h uint16.h \
 ndelay.h socket.h uint16.h droproot.h qlog.h uint16.h response.h \
 uint32.h dns.h stralloc.h gen_alloc.h iopause.h taia.h tai.h uint64.h \
 taia.h iopause.h alloc.h str.h
diff --git a/dnscache.c b/dnscache.c
index abcba69..106ae0b 100644
--- a/dnscache.c
+++ b/dnscache.c
@@ -396,6 +396,7 @@ int main()
 {
   char *x;
   unsigned int i, j, k;
+  unsigned long port = 53;
   unsigned long cachesize;
   static stralloc sa = {0};
 
@@ -415,16 +416,19 @@ int main()
     noipv6 = 1;
 #endif
 
+  x = env_get("PORT");
+  if (x) scan_ulong(x,&port);
+
   udp53 = socket_udp6();
   if (udp53 == -1)
     strerr_die2sys(111,FATAL,"unable to create UDP socket: ");
-  if (socket_bind6_reuse(udp53,myipincoming,53,interface) == -1)
+  if (socket_bind6_reuse(udp53,myipincoming,(uint16) port,interface) == -1)
     strerr_die2sys(111,FATAL,"unable to bind UDP socket: ");
 
   tcp53 = socket_tcp6();
   if (tcp53 == -1)
     strerr_die2sys(111,FATAL,"unable to create TCP socket: ");
-  if (socket_bind6_reuse(tcp53,myipincoming,53,interface) == -1)
+  if (socket_bind6_reuse(tcp53,myipincoming,(uint16) port,interface) == -1)
     strerr_die2sys(111,FATAL,"unable to bind TCP socket: ");
 
   droproot(FATAL);
diff --git a/server.c b/server.c
index d52ce87..8fbd886 100644
--- a/server.c
+++ b/server.c
@@ -1,6 +1,7 @@
 #include "byte.h"
 #include "case.h"
 #include "env.h"
+#include "scan.h"
 #include "buffer.h"
 #include "strerr.h"
 #include "ip4.h"
@@ -92,10 +93,14 @@ int main()
 {
   char *x;
   int *udp53;
+  unsigned long udp53_port = 53;
   unsigned int off;
   unsigned int cnt;
   iopause_fd *iop;
 
+  x = env_get("PORT");
+  if (x) scan_ulong(x,&udp53_port);
+
   x = env_get("IP");
   if (!x)
     strerr_die2x(111,fatal,"$IP not set");
@@ -138,7 +143,7 @@ int main()
         ifid=socket_getifidx(interface);
       l+=Len;
     }
-    if (socket_bind6_reuse(udp53[cnt],ip,53,ifid) == -1)
+    if (socket_bind6_reuse(udp53[cnt],ip,(uint16) udp53_port,ifid) == -1)
       strerr_die2sys(111,fatal,"unable to bind UDP socket: ");
     ndelay_off(udp53[cnt]);
     socket_tryreservein(udp53[cnt],65536);
